inc/indexables/validators/class-object-type-validator.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Internals
+ */
+
+/**
+ * Class WPSEO_Object_Type_Validator
+ */
+class WPSEO_Object_Type_Validator implements WPSEO_Endpoint_Validator {
```

Comment:
We could optimize this class by using individual classes for the different object types. We have different `WPSEO_Indexable` classes for that same purpose, so a similar abstraction would make sense here as well.

The `WPSEO_Object_Type_Validator` class could hold multiple sub-classes extending a `WPSEO_Object_Type` base class which has `get_type()`, `get_subtypes()`, and `is_subtype( $subtype )` methods. The default implementation would have `WPSEO_Post_Object_Type` and `WPSEO_Term_Object_Type`, but those could be filterable to add more.